tests/lib/rules/prefer-stateless-function.js

```diff
@@ -513,6 +513,289 @@ ruleTester.run('prefer-stateless-function', rule, {
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should work without render function
+      code: [
+        'class Foo extends Component {',
+        '}'
+      ].join('\n'),
+      output: [
+        'let Foo = (props) => {};'
+      ].join('\n'),
+      parser: 'babel-eslint',
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should leave as it is when class is not named
+      code: [
+        'let x = class extends Component {',
+        '}'
+      ].join('\n'),
+      output: [
+        'let x = class extends Component {',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint',
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should just change stateful to stateless for default parser
+      code: [
+        'class Foo extends Component {',
+        '  render() {',
+        '    return false;',
+        '  }',
+        '}'
+      ].join('\n'),
+      output: [
+        'let Foo = (props) => {',
+        '  return false;',
+        '};'
+      ].join('\n'),
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should rename this.props into just props for default parser
+      code: [
+        'class Foo extends Component {',
+        '  render() {',
+        '    const { foo, bar } = this.props;',
+        '',
+        '    return <div>{this.props.test}</div>;',
+        '  }',
+        '}'
+      ].join('\n'),
+      output: [
+        'let Foo = (props) => {',
+        '  const { foo, bar } = props;',
+        '',
+        '  return <div>{props.test}</div>;',
+        '};'
+      ].join('\n'),
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should keep formatting of source code for default parser
+      code: [
+        'class Foo extends Component {',
+        '  render() {',
+        '    const {foo, bar} = this.props;',
+        '    ',
+        '    return <div>',
+        '       {this.props.test}</div>;',
+        '  }',
+        '}',
+        'Foo.propTypes={foo: PropTypes.func};',
+        'Foo.displayName=\'Bar\';',
+        'Foo.foo=\'Baz\';'
+      ].join('\n'),
+      output: [
+        'let Foo = (props) => {',
+        '  const {foo, bar} = props;',
+        '  ',
+        '  return <div>',
+        '     {props.test}</div>;',
+        '};',
+        'Foo.propTypes={foo: PropTypes.func};',
+        'Foo.displayName=\'Bar\';',
+        'Foo.foo=\'Baz\';'
+      ].join('\n'),
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      // should work without render function for default parser
+      code: [
+        'class Foo extends Component {',
+        '}'
+      ].join('\n'),
+      output: [
+        'let Foo = (props) => {};'
```

Comment:
Why does the fixer output an arrow function for SFCs?

This will run afoul of, for example, the airbnb style guide, which only allows named functions as SFCs. If we instead defaulted to a normal named function here, other rules would be able to convert it to an SFC.

Also, converting from a `class` declaration to an expression isn't always a safe change; the only safe replacement for `class` is a `function` declaration (which will be an expression when in expression position).